daemons/controld/controld_cib.c

```diff
@@ -22,6 +22,90 @@
 // Call ID of the most recent in-progress CIB resource update (or 0 if none)
 static int latest_rsc_update = 0;
 
+// Call IDs of requested CIB replacements that won't trigger a new election
+// (used as a set of gint values)
+static GHashTable *cib_replacements = NULL;
+
+/*!
+ * \internal
+ * \brief Store the call ID of a CIB replacement that the controller requested
+ *
+ * The \p do_cib_replaced() callback function will avoid triggering a new
+ * election when we're notified of one of these expected replacements.
+ *
+ * \param[in] call_id  CIB call ID (or 0 for a synchronous call)
+ *
+ * \note This function should be called after making any asynchronous CIB
+ *       request (or before making any synchronous CIB request) that may replace
+ *       part of the nodes or status section. This may include CIB sync calls.
+ */
+void
+controld_record_cib_replace_call(int call_id)
+{
+    CRM_CHECK(call_id >= 0, return);
+
+    if (cib_replacements == NULL) {
+        cib_replacements = g_hash_table_new(NULL, NULL);
+    }
+
+    /* If the call ID is already present in the table, then it's old. We may not
+     * be removing them properly, and we could improperly ignore replacement
+     * notifications if cib_t:call_id wraps around.
+     */
+    CRM_LOG_ASSERT(g_hash_table_add(cib_replacements,
+                                    GINT_TO_POINTER((gint) call_id)));
+}
+
+/*!
+ * \internal
+ * \brief Remove the call ID of a CIB replacement from the replacements table
+ *
+ * \param[in] call_id  CIB call ID (or 0 for a synchronous call)
+ *
+ * \return \p true if \p call_id was found in the table, or \p false otherwise
+ *
+ * \note CIB notifications run before CIB callbacks. If this function is called
+ *       from within a callback, \p do_cib_replaced() will have removed
+ *       \p call_id from the table first if relevant changes triggered a
+ *       notification.
+ */
+bool
+controld_remove_cib_replace_call(int call_id)
```

Comment:
maybe s/remove/forget/ since the calls aren't actually affected

Reply:
Sounds good